File: zfrontend/src/models/donation.ts
import { BasePathV1 } from "@/requests/base";

export class Donation {
    constructor(
        public id: string,
        public firstName: string,
        public lastName: string,
        public createdAtUtc: number,
        public amount: number,
        public thankYouComment: string,
        public isAnonymous: boolean,
        public companyName: string,
        public __typename: string
    ) { }
}

// TransactionWithDonation class est utilisé comme modèle et pour récupérer les donations
export class TransactionWithDonation {
    constructor(
        public id: string,
        public type: string,
        public refundedAmount: number,
        public donation: Donation,
        public __typename: string
    ) { }

    static async getAll(page: number, pageSize: number = 10): Promise<{ data: TransactionWithDonation[], hasNextPage: boolean }> {
        const headers: Headers = new Headers();
        headers.set("Content-Type", "application/json");
        headers.set("Accept", "application/json");

        const request: Request = new Request(BasePathV1 + `/donations?page=${page}&pageSize=${pageSize}`, {
            method: "GET",
            headers,
        });

        const response = await fetch(request);
        if (!response.ok) {
            throw new Error(response.statusText);
        }

        const json = await response.json();
        if (json.errors) {
            throw new Error(json.errors[0].message);
        }

        return json;
    }
}

File: backend_go/data/models/donation.go
package models

type TransactionWithDonationObject struct {
	Id             *string         `json:"id"`
	Type           *string         `json:"type"`
	RefundedAmount *int            `json:"refundedAmount"`
	Donation       *DonationObject `json:"donation"`
	Typename       *string         `json:"__typename"`
}

type DonationObject struct {
	Id              *string `json:"id"`
	FirstName       *string `json:"firstName"`
	LastName        *string `json:"lastName"`
	CreatedAtUtc    *int    `json:"createdAtUtc"`
	Amount          *int    `json:"amount"`
	ThankYouComment *string `json:"thankYouComment"`
	IsAnonymous     *bool   `json:"isAnonymous"`
	CompanyName     *string `json:"companyName"`
	Typename        *string `json:"__typename"`
}
